### PR TITLE
Add varbinary and binary support for DB2 - fixes #3408 

### LINKED ISSUE #3408: VARBINARY datatype is wrongly created as BLOB on DB2 z/OS
## Environment

**Liquibase Version**: 4.12.0

**Liquibase Integration & Version**:  Spring Boot 1.5.22.RELEASE

**Liquibase Extension(s) & Version**:  liquibase-slf4j 4.1.0

**Database Vendor & Version**:  DB2 z/OS 12

**Operating System Type & Version**: RHEL 7

**Infrastructure Type/Provider**: VM

## Description
Given a changeset that creates a new VARBINARY column, the column is created as BLOB although VARBINARY is a supported datatype by DB2 z/OS.
DB2 z/OS 12 reference: https://www.ibm.com/docs/en/db2-for-zos/12?topic=programs-data-types-columns
DB2 z/OS 11 reference: https://www.ibm.com/docs/en/db2-for-zos/11?topic=programs-data-types-columns

## Steps To Reproduce

Using spring boot and given the next change set:
```xml
<changeSet id="1" author="author">
        <createTable tableName="MyTable" tablespace="tableSpace">
            <column name="ID" type="VARCHAR(16)">
                <constraints nullable="false" primaryKey="true" primaryKeyName="PK_MyTable"/>
            </column>
            <column name="COLUMN_A" type="VARBINARY(4096)"/>
            <column name="COLUMN_B" type="VARBINARY(255)"/>
        </createTable>
</changeSet>
```

## Actual Behavior
COLUMN_A and COLUMN_B  columns are created in DB2 as BLOB datatype.
## Expected/Desired Behavior
COLUMN_A and COLUMN_B  columns are created in DB2 as VARBINARY datatype.

### PR DESCRIPTION
**Pull request by mihaelaDev**
## Impact

- [X ] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Add varbinary and binary support for DB2. As mentioned in issue #3408 , these are supported data types in DB2.
- Closes #3408

## Things to be aware of

-Adjusted liquibase.datatype.core.BlobType#toDatabaseDataType method.
-Introduced junit-jupiter-params test dependency for the testing

## Things to worry about

- Not sure if there are other integrations tests to be adjusted
- Not sure how the max value is handled in case it's exceeded.

### CODE REVIEW

**Comment by nvoxland** [comment]
**Pre-Review Notes**

Overall changes make sense off hand. The new `junit-jupiter-params` lets the existing junit test be parameter driven without having to change to spock, it looks like.

Questions I have:
- The jupiter-parms doesn't force an upgrade to the version of junit we are using?

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Review by filipelautert — APPROVED** [review]
* @nvoxland - according to the documentation, JUnit 4 and 5 can be executed together since surefire 2.22 (we are on 2.22.2) -> https://junit.org/junit5/docs/snapshot/user-guide/#running-tests-build-maven , so we are fine ;)
* Change affects only DB2 databases; tests are passing
* Tests added and passing
* At first I though that the if logic from BlobType could be joined with the H2/HSQLDB if, but this could add some confusion and even more ifs. 
* @mihaelaDev thanks for your contribution!

File: liquibase-core/src/test/groovy/liquibase/datatype/DataTypeFactoryTest.groovy [hunk 1]
```diff
@@ -33,6 +33,8 @@ class DataTypeFactoryTest extends Specification {
         "serial4"                                      | new MockDatabase()     | "INT"                                          | IntType       | true
         "xml"                                          | new MockDatabase()     | "XML"                                          | XMLType       | false
         "real"                                         | new DB2Database()      | "REAL"                                         | FloatType     | false
+        "varbinary(200)"                               | new DB2Database()      | "VARBINARY(200)"                               | BlobType      | false
```
Comment (XDelphiGrl) [inline liquibase-core/src/test/groovy/liquibase/datatype/DataTypeFactoryTest.groovy:36]: @filipelautert, hello - Can we add another test for this case?

```
originalDefinition.startsWith("java.sql.Types.VARBINARY")) 
```

**Review by XDelphiGrl — APPROVED** [review]
This PR fixes a bug a generate SQL bug where the DB2 `VARBINARY` datatype was incorrectly generated as `BLOB`. 
* New integration tests added for various permutations of `VARBINARY`, `BINARY` and `BLOB`.
* Automated functional and test harness executions are passing.
* No additional testing required.

APPROVED

**Review by suryaaki2 — APPROVED** [review]

**Comment by mihaelaDev** [comment]
Hello, any idea when the new release with this fix will be done?

**Comment by filipelautert** [comment]
Hi @mihaelaDev! We plan to release anytime in January. Due to end of the year things got a bit slower than usual on our side .

**Comment by scordio** [comment]
Hi @filipelautert, I would also be interested in using this fix.

Do you have any rough dates in mind? No pressure, of course, just to have a feeling and thanks a lot for taking care of it!

**Comment by filipelautert** [comment]
Hi @scordio , no problems :) I believe that by the end of next week we will have it released .

**Pull request merged**